hdf5/cppbuild.sh

```diff
@@ -9,21 +9,45 @@ fi
 
 ZLIB=zlib-1.2.13
 HDF5_VERSION=1.14.0
+AEC_VERSION=1.0.6
 download "http://zlib.net/$ZLIB.tar.gz" $ZLIB.tar.gz
 download "https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.14/hdf5-$HDF5_VERSION/src/hdf5-$HDF5_VERSION.tar.bz2" hdf5-$HDF5_VERSION.tar.bz2
+#download "https://gitlab.dkrz.de/k202009/libaec/uploads/45b10e42123edd26ab7b3ad92bcf7be2/libaec-$AEC_VERSION.tar.gz" libaec-$AEC_VERSION.tar.gz
+download "https://github.com/MathisRosenhauer/libaec/releases/download/v$AEC_VERSION/libaec-$AEC_VERSION.tar.gz" libaec-$AEC_VERSION.tar.gz
 
 mkdir -p $PLATFORM
 cd $PLATFORM
 INSTALL_PATH=`pwd`
 echo "Decompressing archives..."
 tar --totals -xf ../hdf5-$HDF5_VERSION.tar.bz2
+tar --totals -xf ../libaec-$AEC_VERSION.tar.gz
 cd hdf5-$HDF5_VERSION
 
 #sedinplace '/cmake_minimum_required/d' $(find ./ -iname CMakeLists.txt)
 sedinplace 's/# *cmakedefine/#cmakedefine/g' config/cmake/H5pubconf.h.in
 sedinplace 's/COMPATIBILITY SameMinorVersion/COMPATIBILITY AnyNewerVersion/g' CMakeInstallation.cmake
 sedinplace '/C_RUN (/{N;N;d;}' config/cmake/ConfigureChecks.cmake
 
+# As of 1.14.0 the integrated cmake process for building aec is broken
+# Revisit if this is needed with 1.14.1
+build_aec_szip() {
+    mkdir -p ../libaec-$AEC_VERSION/build
+    cd ../libaec-$AEC_VERSION/build
+    "$CMAKE" $@ -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$INSTALL_PATH ..
+    make -j $MAKEJ
+    make install
+    cd ../../hdf5-$HDF5_VERSION
+}
+build_aec_szip_windows() {
+    mkdir -p ../libaec-$AEC_VERSION/build
+    cd ../libaec-$AEC_VERSION/build
+    "$CMAKE" $@ -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$INSTALL_PATH ..
+    ninja -j $MAKEJ
+    ninja install
+    cd ../../hdf5-$HDF5_VERSION
+}
```

Comment:
Let's not create functions like this. It gets ugly real quick. Place anything in common for all platforms in variables only, and anything else in the switch below.

Reply:
I removed the functions in [aaf057d](https://github.com/bytedeco/javacpp-presets/pull/1334/commits/aaf057dccf347ddd85082e45c21b7d80ce9d7774). I increased use of `pushd` / `popd` to help keep the directories straight.